pkgs/by-name/bu/butler/package.nix

```diff
@@ -0,0 +1,41 @@
+{
+  buildGoModule,
+  brotli,
+  lib,
+  fetchFromGitHub,
+}:
+
+# update instructions:
+# - Check if butler version bug was fixed https://github.com/itchio/butler/issues/266
+# - if it's fixed, remove patch.
+# - if it was not fixed, follow steps below to regenerate the patch
+# - manually clone butler, change go.mod's version number to 1.18 at least
+# - run `go mod tidy` in the cloned repository.
+# - generate patch with `git diff > go.mod.patch`
+
+buildGoModule rec {
+  pname = "butler";
+  version = "15.24.0";
+
+  src = fetchFromGitHub {
+    owner = "itchio";
+    repo = "butler";
+    tag = "v${version}";
+    hash = "sha256-Gzf+8icPIXrNc8Vk8z0COPv/QA6GL6nSvQg13bAlfZM=";
+  };
+
+  buildInputs = [ brotli ];
+
+  patches = [ ./go.mod.patch ];
+
+  doCheck = false; # disabled because the tests don't work in a non-FHS compliant environment.
+
+  vendorHash = "sha256-A6u7bKI7eoptkjBuXoQlLYHkEVtrl8aNnBb65k1bFno=";
+
+  meta = {
+    description = "Command-line itch.io helper";
+    homepage = "http://itch.io";
```

Comment:
add changelog and platforms

Reply:
I've been informed platforms should be left empty so that it inherits buildGoModule's defaults (defaults to platforms that Go supports).

I've dug around for commits for changelog changes and I can't find anything ([previous butler init](https://github.com/NixOS/nixpkgs/commit/e9b39ae01c1af6f4be2359a19c0817656412d82c) has no changelog). I'm not sure what you're asking for, I'm sorry!

Reply:
> I've been informed platforms should be left empty so that it inherits buildGoModule's defaults (defaults to platforms that Go supports).

Good to know. Thanks.

> I've dug around for commits for changelog changes and I can't find anything ([previous butler init](https://github.com/NixOS/nixpkgs/commit/e9b39ae01c1af6f4be2359a19c0817656412d82c) has no changelog). I'm not sure what you're asking for, I'm sorry!

https://github.com/itchio/butler/releases/tag/v15.24.0/CHANGELOG.md . And then you can replace the version with your version here.

Reply:
Oh! I see, I found the relevant part in the manual. Thanks for the help! I've added the changelog meta element as requested.